src/compiler/checker.ts

```diff
@@ -17880,6 +18186,26 @@ namespace ts {
             return arrayElementType;
         }
 
+        function getIteratedTypeOfIterableOrElementTypeOfArray(inputType: Type, errorNode: Node, checkAssignability: boolean): Type {
+            const iteratedType = compilerOptions.downlevelIteration && getIteratedTypeOfIterable(inputType, /*errorNode*/ undefined);

```

Comment:
`getIteratedTypeOfIterableOrElementTypeOfArrayOrString` has different logic than `getIteratedTypeOfIterableOrElementTypeOfArray`, we don't want to delegate.